internal/hcp/internal/controllers/link/controller.go

```diff
@@ -115,10 +115,17 @@ func (r *linkReconciler) Reconcile(ctx context.Context, rt controller.Runtime, r
 	}
 
 	if resource.IsMarkedForDeletion(res) {
-		if err = cleanup(ctx, rt, res, r.dataDir); err != nil {
+		if err = cleanup(rt, r.hcpManager, r.dataDir); err != nil {
 			rt.Logger.Error("error cleaning up link resource", "error", err)
 			return err
 		}
+
+		err := ensureDeleted(ctx, rt, res)
+		if err != nil {
+			rt.Logger.Error("error deleting link resource", "error", err)
+
+			return err
+		}
```

Comment:
I know the plan is to remove the finalizer later, but maybe we should just remove it now and only rely on `cleanup` being called in the `NotFound` case above. Since we've learned it's not providing us any value in terms of resiliency like we thought, it just makes reasoning about this code more complicated until we're able to change it.

Reply:
My understanding is that the Finalizer still provides more resiliency compared to the standard delete pattern in the reconciler, but just not across multiple controllers (i.e., placement on each server). So when we change to have a singleton controller on just the leader, we may still want to use a Finalizer.

I think it'll come down to how much cleanup the controller will actually end up doing after we make the placement changes. Like if we keep the config directory cleanup in the controller, then we may want to have a Finalizer. But if the manager is going to do all the cleanup, then a Finalizer might not be necessary since there's not really anything to be interrupted unexpectedly in a reconciliation.

Reply:
That was a long winded way of saying that I lean towards keeping it for now and removing it once it's no longer giving us any advantages! But let me know what you think!

Reply:
I assumed we would be moving all the cleanup into the HCP manager, but I'm okay with waiting and seeing where we end up because you're right, if we do have cleanup in the controller still, we'd want the finalizer.